airflow/sensors/smart_sensor.py

```diff
@@ -672,13 +672,13 @@ def _execute_sensor_work(self, sensor_work):
 
     def flush_cached_sensor_poke_results(self):
         """Flush outdated cached sensor states saved in previous loop."""
-        for key, cached_work in self.cached_dedup_works.items():
+        for key, cached_work in self.cached_dedup_works.copy().items():
```

Comment:
hmm Why do we need `copy()`?

Reply:
Dictionaries are represented by a hash table and adding or removing items while iterating over it will alter the iteration order. This will cause a RuntimeError.

If you need to add items to the dictionary during iteration, it is recommended to iterate over a shallow `copy` of the dictionary.